src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,83 @@
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList: React.FC<Props> = React.memo(
+  ({ todoList, onEyeClick, isModalOpen }) => {
+    const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+    const handleEyeClick = useCallback(
+      (todoId: number, userId: number) => {
+        setSelectedTodoId(prevSelectedTodoId =>
+          prevSelectedTodoId === todoId ? null : todoId,
+        );
+        onEyeClick(todoId, userId);
+      },
+      [onEyeClick],
+    );
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+    TodoList.displayName = 'TodoList';
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-    </tbody>
-  </table>
+    return (
+      <table className="table is-narrow is-fullwidth">
+        <thead>
+          <tr>
+            <th>#</th>
+            <th>
+              <span className="icon">
+                <i className="fas fa-check" />
+              </span>
+            </th>
+            <th>Title</th>
+            <th> </th>
+          </tr>
+        </thead>
+
+        {todoList.map(todo => (
+          <tbody key={todo.id}>
+            <tr data-cy="todo" className="">
```

Comment:
you need to map the `tr`
<img width="377" alt="image" src="https://github.com/user-attachments/assets/18434d4c-d548-405c-9cce-db124be8cca6">
instead of tbody